vassal-app/src/main/java/VASSAL/build/module/map/MovementReporter.java

```diff
@@ -313,6 +315,10 @@ public Point getOldPosition() {
       return oldPosition;
     }
 
+    public List<GamePiece> getPieces() {
+      return new ArrayList<>(pieces);
+    }
+
```

Comment:
That kills me that you have to copy the whole list just to get the first element.

Reply:
Even more hilarious is that, as far as I can tell, we NEVER ACTUALLY PUT MORE THAN ONE PIECE IN THIS!!!

Reply:
Which puts me into this newer/better idea -- I changed it to something that just returns the first piece in the array. That will at least save all the making-a-copy. Sometimes I think too much in terms of "future extensibility"

Reply:
So anyway that's checked in now.

Reply:
Why the first piece? Why not the last piece or the second piece? This is the reason that Map level auto-reports never used GamePiece properties, they are meaningless if you are dragging more than one piece.

Reply:
Well that's what I *thought* except then Benkyo pointed out that it never actually does that. It generates one "MoveSummary" for EACH piece moved.

Reply:
Or, well, no.

Reply:
So yes ... if we _want_ to make this work and "non-dodgy", then I'd probably put a checkbox on there telling the Map to report separately for every unit (and not group a single stack together in one autoreport line). And so then if you checked that box we could use the piece as the propertysource for each report. 

We can discuss whether it makes sense to go ahead and do that, or if we should leave folks who want "complex move reporting" to create their own logic in the pieces themselves (which is completely possible - just the "on ramp" from these things is a little hard to find)